### PR TITLE
LPS-151632 Move petra-portlet-url-builder to apps/portal

### PR DESCRIPTION
**Pull request by julschong**
@tinatian 
Hi Tina, 
This is resubmit from #1227 to separate out the big usage commit into smaller chunks.

Thank you!

### CODE REVIEW

**Comment by catarineaguiar** [comment]
@julschong code conflicts, can you rebase, please?

**Comment by julschong** [comment]
rebased to resolve conflict in 

> modules/apps/document-library/document-library-web/build.gradle

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
@julschong any test ran on CI, I think it was the same problem we had [here](https://github.com/liferay-core-infra/liferay-portal/pull/1173), can you take a look? Thanks

**Comment by julschong** [comment]
@catarineaguiar Hi Catarine,
This pull is quite different from #1173
and this was running ci before i did rebase.
Let's trigger relevant again and see

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by julschong** [comment]
rebased again

**Comment by julschong** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
@julschong no regressions on relevant. Can you rebase please? Thanks

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
rebased with 4 new usages added

File: modules/core/portal-bootstrap/system.packages.extra.bnd [hunk 1]
```diff
@@ -290,12 +290,12 @@ Provide-Capability:\
...
 	@lib/com.liferay.petra.nio.jar,\
-	@lib/com.liferay.petra.portlet.url.builder.jar,\
 	@lib/com.liferay.petra.process.jar,\
 	@lib/com.liferay.petra.reflect.jar,\
 	@lib/com.liferay.petra.sql.dsl.api.jar,\
 	@lib/com.liferay.petra.sql.dsl.spi.jar,\
 	@lib/com.liferay.petra.string.jar,\
 	@lib/com.liferay.petra.url.pattern.mapper.jar,\
-	@lib/com.liferay.petra.xml.jar
+	@lib/com.liferay.petra.xml.jar,\
```
Comment (tinatian) [inline modules/core/portal-bootstrap/system.packages.extra.bnd:299]: @julschong The commit message is not consistent with the change in this commit, please update.

Tina

**Comment by julschong** [comment]
updated commit message [f75ed9a](https://github.com/liferay-core-infra/liferay-portal/pull/1281/commits/f75ed9af608774a99506bf974b96e420ab02e80a)

**Comment by catarineaguiar** [comment]
waiting for relevant results.

**Comment by catarineaguiar** [comment]
@julschong no regression on relevant. Rebase please :)

**Comment by julschong** [comment]
@catarineaguiar 
rebased again there were many conflicts so I will run relevant again

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
@julschong formatting issue:

```
 [java] java.lang.Exception: Found 2 formatting issues:
     [java] 1: Unneeded/incorrect Export-Package: com.liferay.portal.upgrade.step.util, see https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/check/bnd_exports_check.markdown: ./modules/apps/static/portal/portal-upgrade-api/bnd.bnd (SourceCheck:BNDExportsCheck)
     [java] 2: ./modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/index.html expected:<... name="viewport" />
     [java] [
     [java] 		<link href="https://cdn.jsdelivr.net/npm/@clayui/css/lib/css/atlas.css" rel="stylesheet" />
     [java] ]
     [java] 		<title>Partner Po...> but was:<... name="viewport" />
     [java] [		<link href="https://cdn.jsdelivr.net/npm/@clayui/css/lib/css/atlas.css" rel="stylesheet" />]
     [java] 		<title>Partner Po...>
     [java] 
     [java] 	at com.liferay.source.formatter.SourceFormatter.format(SourceFormatter.java:442)
     [java] 	at com.liferay.source.formatter.SourceFormatter.main(SourceFormatter.java:284)
```
full log [here](https://testray.liferay.com/reports/production/logs/2022-08/test-1-27/test-portal-source-format/3006/source-format-current-branch-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by catarineaguiar** [comment]
no other regression.

**Comment by tinatian** [comment]
@julschong Please fix the format issue.

Tina.

**Comment by julschong** [comment]
rebased

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
rebased again

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
@catarineaguiar 
HI Catarine,

SF issue is from upstream and unrelated to my code change.
Seems that they just haven't fixed it yet

Thank you!

**Comment by catarineaguiar** [comment]
@julschong ok, thanks for checking. Rebase and run relevant if necessary, please.

**Comment by julschong** [comment]
rebased

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:stop

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
@julschong re-running relevant because testray had problems to generate the results.

**Comment by tinatian** [comment]
@julschong This pull is easy to get code conflicts, please try to separate into small pieces.

Thanks.
Tina

**Pull request closed without merging**